bodhi/tests/server/consumers/test_pungi_wrapper.py

```diff
@@ -0,0 +1,419 @@
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program; if not, write to the Free Software
+# Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+
+import mock
+import pytest
+import os
+
+
+from bodhi.server.consumers.pungi_wrapper import (
+    PungiWrapper, PungiConfig, VariantsConfig
+)
+
+from pungi.compose import Compose
+
+
+@pytest.mark.usefixture("get_pungi_conf")
+@pytest.fixture(scope="module")
+def get_compose(pungi_conf, compose_dir):
+
+    logger = mock.Mock()
+
+    compose = Compose(pungi_conf, topdir=compose_dir, debug=True, skip_phases=["productimg"],
+                      just_phases=[], old_composes=None, koji_event=None, supported=False,
+                      logger=logger, notifier=mock.Mock())
+
+    return compose
+
+
+@pytest.fixture(scope="module")
+def get_pungi_conf(tmpdir, config_str=None):
+    conf = tmpdir.mkdir("conf").join("pungi.conf")
+    config_data = """ # RELEASE
+    release_name = "Fedora"
+    release_short = "Fedora"
+    release_version = "23"
+
+    # GENERAL SETTINGS
+    comps_file = "comps-f23.xml"
+    variants_file = "variants-f23.xml"
+
+    # KOJI
+    koji_profile = "koji"
+    runroot = False
+
+    # PKGSET
+    sigkeys = [None]
+    pkgset_source = "koji"
+    pkgset_koji_tag = "f23"
+
+    # CREATEREPO
+    createrepo_checksum = "sha256"
+
+    # GATHER
+    gather_source = "comps"
+    gather_method = "deps"
+    greedy_method = "build"
+    check_deps = False
+
+    # BUILDINSTALL
+    bootable = True
+    buildinstall_method = "lorax"
+    release_is_layered = True
+
+    base_product_name = "Fedora"
+    base_product_short = "Fedora"
+    base_product_version = "23" """
+    if config_str:
+        config_data = config_str
+    conf.write(config_data)
+
+    return conf
+
+
+@pytest.mark.usefixture("get_pungi_conf")
+class TestPungiConfig(object):
+
+    def test_load_config_from_file(self, tmpdir):
+        logger = mock.Mock()
+        pungi_conf = get_pungi_conf(tmpdir)
+
+        conf_xml = PungiConfig(path=str(pungi_conf), logger=logger)
+        assert isinstance(conf_xml, dict)
+        assert isinstance(conf_xml["sigkeys"], list)
+        assert conf_xml["release_name"] == "Fedora"
+        assert conf_xml["release_version"] == "23"
+
+    def test_valid_pungi_config(self, tmpdir):
+        logger = mock.Mock()
+        config_data = """ # RELEASE
+        release_name = "Fedora"
+        release_short = "Fedora"
+        release_version = "23" """
+        pungi_conf = get_pungi_conf(tmpdir, config_data)
+
+        with pytest.raises(Exception):
+            PungiConfig(path=str(pungi_conf), logger=logger)
```

Comment:
Let's make an assertion about the Exception that is raised here, like its type and its message.